src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/FakeRoleAssignmentsGenerator.java

```diff
@@ -0,0 +1,104 @@
+package uk.gov.hmcts.ccd.domain.service.casedataaccesscontrol;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.data.definition.CachedCaseDefinitionRepository;
+import uk.gov.hmcts.ccd.data.definition.CaseDefinitionRepository;
+import uk.gov.hmcts.ccd.data.user.CachedUserRepository;
+import uk.gov.hmcts.ccd.data.user.UserRepository;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignment;
+import uk.gov.hmcts.ccd.domain.model.definition.UserRole;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Stream.concat;
+
+@Component
+@ConditionalOnProperty(name = "enable-attribute-based-access-control", havingValue = "true")
+public class FakeRoleAssignmentsGenerator {
+
+    private static final Pattern RESTRICT_GRANTED_ROLES_PATTERN =
+        Pattern.compile(".+-solicitor$|.+-panelmember$|^citizen(-.*)?$|^letter-holder$|^caseworker-.+-localAuthority$");
+    public static final String IDAM_PREFIX = "idam:";
+    public static final String GRANT_TYPE_SPECIFIC = "SPECIFIC";
+    public static final String GRANT_TYPE_STANDARD = "STANDARD";
+    public static final String CLASSIFICATION_RESTRICTED = "RESTRICTED";
+    public static final String ROLE_TYPE_CASE = "CASE";
+
+    private final UserRepository userRepository;
+    private final CaseDefinitionRepository caseDefinitionRepository;
+
+    @Autowired
+    public FakeRoleAssignmentsGenerator(@Qualifier(CachedUserRepository.QUALIFIER) UserRepository userRepository,
+                                        @Qualifier(CachedCaseDefinitionRepository.QUALIFIER)
+                                            CaseDefinitionRepository caseDefinitionRepository) {
+        this.userRepository = userRepository;
+        this.caseDefinitionRepository = caseDefinitionRepository;
+    }
+
+    public List<RoleAssignment> addFakeRoleAssignments(List<RoleAssignment> roleAssignments) {
+        List<String> idamUserRoles = new ArrayList<>(userRepository.getUserRoles());
+        List<String> grantedRoles = filterGrantedRoles(idamUserRoles);
+        List<RoleAssignment> augmentedRoleAssignments;
+
+        if (!grantedRoles.isEmpty()) {
```

Comment:
I think this might work, but double check please:
```suggestion
        if (caseAccessService.canOnlyViewExplicitlyGrantedCases()) {
```
I think It's less code duplication because the logic you wrote is already in canOnlyViewExplicitlyGrantedCases()
see: https://github.com/hmcts/ccd-data-store-api/blob/80adc76067063ba3c3600fb3e0674b41bfe5426f/src/main/java/uk/gov/hmcts/ccd/domain/service/common/CaseAccessService.java#L116